x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/SpatialPlugin.java

```diff
@@ -205,4 +210,10 @@ private static void registerCardinalityAggregator(ValuesSourceRegistry.Builder b
             return realParser.parse(parser, name);
         };
     }
+
+    @Override
+    public void loadExtensions(ExtensionLoader loader) {
+        // we only expect one vector tile extension that comes from the vector tile module.
+        loader.loadExtensions(VectorTileExtension.class).forEach(vectorTileExtension::set);
```

Comment:
That essentially moves the dependency from the compile time into runtime. It is not explicit, but the whole thing will break if spatial module is present but vector-tile module is not or if we have another alternative implementation for that. We should probably address this as a follow up as well.

Reply:
My hope here is that if the vector-tile module is not present everything should just work. You just end up with an error if you request geometries in mvt format.

Reply:
Yes, I stand corrected, it looks like it throws "vector tile format is not supported" exception in this case, which is hardcoded in the spatial module. But I think there is still too much knowledge to vector tile implementation inside spatial, let me try to take a shot at confine more of it into the vector tile module.

Reply:
That would be awesome. I ma going to push it as it is and we can iterate in a follow up.